tests/suites/test_suite_lms.function

```diff
@@ -149,18 +149,40 @@ exit:
 /* END_CASE */
 
 /* BEGIN_CASE */
-void lms_import_export_test (  data_t * pub_key )
+void lms_import_export_test (  data_t * pub_key, int expected_import_rc )
 {
     mbedtls_lms_public_t ctx;
-    uint8_t exported_pub_key[MBEDTLS_LMS_PUBLIC_KEY_LEN(MBEDTLS_LMS_SHA256_M32_H10)];
+    size_t exported_pub_key_buf_size = 0;
+    size_t exported_pub_key_size = 0;
+    unsigned char *exported_pub_key = NULL;
 
     mbedtls_lms_public_init(&ctx);
     TEST_EQUAL( mbedtls_lms_import_public_key( &ctx, pub_key->x, pub_key->len ), 0 );
-    TEST_EQUAL( mbedtls_lms_export_public_key( &ctx, exported_pub_key,
-                                                sizeof(exported_pub_key), NULL ), 0 );
 
-    ASSERT_COMPARE( pub_key->x, MBEDTLS_LMS_PUBLIC_KEY_LEN(MBEDTLS_LMS_SHA256_M32_H10),
-                    exported_pub_key, MBEDTLS_LMS_PUBLIC_KEY_LEN(MBEDTLS_LMS_SHA256_M32_H10) );
+    if( expected_import_rc == 0 )
+    {
+        exported_pub_key_buf_size = MBEDTLS_LMS_PUBLIC_KEY_LEN(MBEDTLS_LMS_SHA256_M32_H10);
+        ASSERT_ALLOC( exported_pub_key, exported_pub_key_buf_size );
+
+        TEST_EQUAL( mbedtls_lms_export_public_key( &ctx, exported_pub_key,
+                                                   exported_pub_key_buf_size,
+                                                   &exported_pub_key_size ), 0 );
+
+        TEST_EQUAL( exported_pub_key_buf_size, exported_pub_key_size );
+        ASSERT_COMPARE( pub_key->x, MBEDTLS_LMS_PUBLIC_KEY_LEN(MBEDTLS_LMS_SHA256_M32_H10),
+                        exported_pub_key, MBEDTLS_LMS_PUBLIC_KEY_LEN(MBEDTLS_LMS_SHA256_M32_H10) );
```

Comment:
Actually, what we know here is that the first string's length is `pub_key->len`, and the second string's length is `exported_pub_key_size`.

There's no particular reason that the first buffer has the indicated size, other than assuming that the test data matches the expectation in the test code.

By reading the code above, we can see that `exported_pub_key_size == exported_pub_key_buf_size` and `exported_pub_key_buf_size == MBEDTLS_LMS_PUBLIC_KEY_LEN(MBEDTLS_LMS_SHA256_M32_H10)` so the second buffer's size is correct, but it would be easier for the reader to use the obvious size.

Also goes for LMOTS.